ludwig/api.py

```diff
@@ -539,6 +540,16 @@ def on_epoch_end(self, trainer, progress_tracker, save_path):
                         save_path=model_dir,
                     )
 
+                    # Calibrates output feature probabilities on validation set if calibration is enabled.
+                    # Must be done after training, and before final model parameters are saved.
+                    if self.backend.is_coordinator() and validation_set is not None:
```

Comment:
This calibrates on the val set if it exists, which is ok, but if there's no val set, i think calibrating on the training set is better than not calibrating at all. May be there's a message that say something like "calibrating on val set" or "no val set, calibrating on train set instead"